### DISCUSSION TITLE
Bidirectional dictionary format that works with GoldenDict-ng?

### DISCUSSION THREAD

**Discussion by michaelbeijer** [xiaoyifang/goldendict-ng#2024] [Q&A]
Hi everyone,

I am trying to find a dictionary format that works with Goldendict, but which allows **bidirectional dictionaries**. I am looking for a text-based format, so I can edit the files in a text editor without having to compile them every time I change something.

So stuff like:

```
Dutch: betalingsrisico; risico van wanbetaling
English: payment risk; risk of non-payment; default risk 

Notes: 
• The risk of non-payment is the possibility that a customer will not pay for goods or services provided.
• Default risk is the risk a lender takes that a borrower will not make the required payments on a debt obligation, such as a loan, a bond, or a credit card. 
```

There must be a format that allows me to add both directions in the same file?

Otherwise, I need to maintain two separate dictionaries, like this:

**Dictionary 1:**
Dutch = English
betalingsrisico; risico van wanbetaling = payment risk; risk of non-payment; default risk

**Dictionary 2:**
English = Dutch
payment risk; risk of non-payment; default risk = betalingsrisico; risico van wanbetaling

**Comment by 101oi** [comment]
![multi_lang_in_one_file](https://github.com/user-attachments/assets/5ed0c6ef-c777-4cd8-a2fc-b69558ea48a5)
If this is what you mean.  Just save the file in .dsl format.
Note: code page for different lang may be complicate.  I use UTF-8.
